tock/hours/views.py

```diff
@@ -613,6 +616,71 @@ def get_object(self, queryset=None):
             user_id=self.request.user.id)
         return obj
 
+    def clean_task_data(self, float_people_id):
+        # Get Float task data for week corresponding w/ ReportingPeriod
+        # instance.
+        r = get_float_data(
+            endpoint='tasks',
+            params={
+                'weeks': base.FLOAT_API_TASK_WEEKS,
+                'start_day': datetime.datetime.strptime(
+                    self.kwargs['reporting_period'], "%Y-%m-%d"
+                ).date()
+            }
+        )
+        if r.status_code != 200:
+            return None
+        else:
+            task_data = r.json()
+        #From response, derive meta data and tasks for the period.
+        f_start_date = datetime.date(
+            int(task_data['start_yr']), 1, 1) + datetime.timedelta(
+            days=int(task_data['start_doy']
+            )-1)
+        f_end_date = f_start_date + datetime.timedelta(
+            days=(base.FLOAT_API_TASK_WEEKS * 6))
```

Comment:
I've [refactored](https://github.com/18F/tock/pull/635/files#diff-bbf773df48e042c9e0b2f421ddc3ee38R636) this one a bit in #635. Originally, I was getting the start and end date of the period in question from the Float response (hence the `f_...`). Float, of course, was getting it from the Tock reporting period start and end dates supplied in the request. I believe my thinking was to ensure what was ultimately being passed to the template in terms of start and end dates came directly from Float (to avoid any confusion over what period the Float data was regarding). However, I later decided (in #635) that this was redundant and nixed it.

The `6` represents weekdays, as in:
```python
>>> import datetime as dt
>>> a = dt.date(2017, 7, 9) # Sunday, or the standard start day of a reporting period
>>> a + dt.timedelta(days=6)
datetime.date(2017, 7, 15) # Saturday, or standard the last day of a reporting period.
```

The `FLOAT_API_TASK_WEEKS` is a overly verbose variable representing the number of weeks of Float data that should be requested (from 1 - 52). I don't really know why we would want to get more than one week of data (since Tock reporting periods are one week long), but I think this may have been some future proofing in case the length of reporting periods changed? Of course, multiplying 6 by 2 (weeks) doesn't get you to a Saturday if the start date is Sunday. Will give this one a rethink...